fw/sock.c

```diff
@@ -791,7 +796,7 @@ ss_tcp_process_skb(struct sock *sk, struct sk_buff *skb, int *processed)
 		 */
 		BUG_ON(conn == NULL);
 
-		if (SS_CONN_TYPE(sk) & Conn_Stop) {
+		if (SS_CONN_TYPE(sk) & Conn_Stop || r) {
```

Comment:
`r` not reinitialized on beginning of iteration. skb on current iteration might be freed relies on result of previous iteration.

Reply:
Yes and it is true. Previously in case of error we goto out and immediately return. But we can't do it because we should correctly update `tp->copied_seq`.  Now we continue to update  `tp->copied_seq` on each iteration after error but stop
processing any skb, because we catch error on previous iteration.